### DISCUSSION TITLE
Homework 5 DrawerValidationTest.SetFromImage

### DISCUSSION THREAD

**Discussion by ianwong1999** [cpp-for-yourself/lectures-and-homeworks#65] [Homeworks Q&A] [answered]
Hi! Thanks for the exciting homework. 

I failed to pass the drawer test:

```
/tmp/homework_checker/pixelator_27bf2274c98672c35010775fb1249d61/validation_tests/drawer_test.cpp:27: Failure
Expected equality of these values:
  drawer.ToString()
    Which is: "\x1B[39m\x1B[40m  \x1B[39m\x1B[49m"
  std::string{"\x1B[39m\x1B[40m \x1B[39m\x1B[49m"}
    Which is: "\x1B[39m\x1B[40m \x1B[39m\x1B[49m"
```

I tried doing the same as the validation test and it seems that, although it has an extra space, it does print two black pixels followed by two red pixels. Are there any configs that I am missing to set up the ftxui::Screener?

**Comment by niosus** [comment]
Hmm. Interesting. Do you know why there is an extra space there?

**Reply by niosus** [reply]
@ianwong1999 do you still have this problem?

**Reply by ianwong1999** [reply, marked as answer]
sorry for the late reply, I just found out that I misunderstood the `dimension` parameter of `pixelator::Drawer` 's constructor, I thought we had to create a ftxui screen with `width=dimension.y*2`. When I just create the ftxui screen with the given argument without modifying, it passes the test. So while the actual output looks reasonable, maybe they have different implementations for the remaining space for the row internally.  Sorry for the inconvenience haha.

**Reply by niosus** [reply]
No worries! If you have suggestions on how to make this clearer, please do tell and I'll try to add these for the future!

**Reply by niosus** [reply]
I'll mark your reply as an answer to your question, but feel free to reopen if needed.
